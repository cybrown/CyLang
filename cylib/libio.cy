void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
