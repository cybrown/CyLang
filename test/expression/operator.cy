import libio;

int a = 2;
int b = 3;
int c = 60;

double da = 2.0;
double db = 3.0;
double dc = 60.0;

//  Testing precedence
putint(a + b * c); putchar(10);
putint((a + b) * c); putchar(10);

// Testing int ops
putint(a + c); putchar(10);
putint(a - c); putchar(10);
putint(a * c); putchar(10);
putint(a / c); putchar(10);
putint(a % c); putchar(10);
putint(a & c); putchar(10);
putint(a | c); putchar(10);
putint(a << c); putchar(10);
putint(a >> c); putchar(10);

// Testing double ops
putdouble(da + dc); putchar(10);
putdouble(da - dc); putchar(10);
putdouble(da * dc); putchar(10);
putdouble(da / dc); putchar(10);
