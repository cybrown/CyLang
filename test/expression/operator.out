182
300
62
-58
120
0
2
0
62
536870912
0
62.000000
-58.000000
120.000000
0.033333
